include/faunus/energy.h

```diff
@@ -2870,9 +3286,19 @@ return u * conc; // -> kT
         for ( int i = 0; i < (int) spc.groupList().size() - 1; i++ )
             for ( int j = i + 1; j < (int) spc.groupList().size(); j++ )
                 u += pot.g2g(p, *spc.groupList()[i], *spc.groupList()[j]);
+
+        if(u != systemEnergy2(spc, pot, p))
+            std::cout << "systemEnergy != systemEnergy2, " << u << " " << systemEnergy2(spc, pot, p) << endl;
         return u;
     }
 
+    template<class Tspace, class Tenergy, class Tpvec>
+    double systemEnergy2( Tspace &spc, Tenergy &pot, const Tpvec &p )
```

Comment:
Why do you need to make a `systemEnergy2()`? Could you not just add `setSpace()` to the original `systemEnergy()`?